File: easy/17. Find Three Largest Numbers/find_three_largest_numbers.cpp
#include <vector>

std::vector<int> findThreeLargestNumbers(std::vector<int> array) {
  int first = INT_MIN;
  int second = INT_MIN;
  int third = INT_MIN;

  for (int i = 0; i < array.size(); i++) {
    if (array[i] > first) {
      third = second;
      second = first;
      first = array[i];
    } else if (array[i] > second) {
      third = second;
      second = array[i];
    } else if (array[i] > third) {
      third = array[i];
    }
  }

  return std::vector<int>{third, second, first};
}

File: easy/17. Find Three Largest Numbers/find_three_largest_numbers.go
package main

var INT_MIN int = -2147483648

func FindThreeLargestNumbers(array []int) []int {
  var first = INT_MIN
  var second= INT_MIN
  var third = INT_MIN

  for _, value := range array {
    if value > first {
      third = second
      second = first
      first = value
    } else if value > second {
      third = second
      second = value
    } else if value > third {
      third = value
    }
  }

  return []int{third, second, first}

}

